### ISSUE TITLE
Dmypy ignoring aditional section names on config file

### ISSUE THREAD

**Issue by tamercuba** [python/mypy#10612]
**Bug Report**

Running dmypy as 
```
python -m mypy.dmypy --status-file '...' run  -- src/ --show-column-numbers --no-error-summary --no-pretty --no-color-output --config-file mypy.ini --python-executable
```
with a `mypy.ini` with the following config:
```
[mypy]
files = sales_register/
python_version = 3.9
ignore_missing_imports = True
follow_imports = skip
plugins = pydantic.mypy
local_partial_types = False


[mypy-adapters.repositories.postgres.migrations.*]
ignore_errors = True
```

are not ignoring errors in `src/adapters/repositories/postgres/migrations/` files. Running just `mypy --config-file=mypy.ini` works fine.

**To Reproduce**

1. Create a `src` folder, and a `src/a` folder inside
2. Create a `test.py` file inside `src/a` with the following code:
```python
def a(x: str) -> str:
    return x

b: int = a('1') # Obviously mypy should raise something here
```
3. Create a `mypy.ini` file with `ignore_errors=True` in `[mypy-a.*]` to skip error in the `src/a/test.py` file
4. Runs `python -m mypy.dmypy --status-file '...' run  -- src/ --show-column-numbers --no-error-summary --no-pretty --no-color-output --config-file mypy.ini --python-executable`

**Expected Behavior**

Shouldnt raise any error

**Actual Behavior**

Raising TypeError

**Your Environment**

- Mypy version used: 0.812
- Python version used: 3.9
- Operating system and version: macOS 11.3.1

**Comment by florimondmanca** [comment]
@tamercuba I bumped into the same issue and realized I had to have a `__init__.py` file present in the `migrations` directory for `mypy` or `dmypy` to take the config file subsection into account. Hope this helps!
